cropharvest/eo/eo.py

```diff
@@ -231,34 +248,50 @@ def _bounding_box_from_centre(
         max_lat, min_lat = mid_lat + deg_lat, mid_lat - deg_lat
         max_lon, min_lon = mid_lon + deg_lon, mid_lon - deg_lon
 
-        return ee.Geometry.Polygon(
+        return (min_lon, min_lat, max_lon, max_lat), ee.Geometry.Polygon(
             [[[min_lon, min_lat], [min_lon, max_lat], [max_lon, max_lat], [max_lon, min_lat]]]
         )
 
+    @classmethod
     def _labels_to_polygons_and_years(
-        self, labels: geopandas.GeoDataFrame, surrounding_metres: int
-    ) -> List[Tuple[ee.Geometry.Polygon, str, int]]:
+        cls, labels: geopandas.GeoDataFrame, surrounding_metres: int
+    ) -> List[Tuple[ee.Geometry.Polygon, str, date, date]]:
 
         output: List[ee.Geometry.Polygon] = []
 
         print(f"Exporting {len(labels)} labels")
 
         for _, row in tqdm(labels.iterrows()):
+            latlons, bounding_box = cls._bounding_box_from_centre(
+                mid_lat=row["lat"], mid_lon=row["lon"], surrounding_metres=surrounding_metres
+            )
+
+            try:
+                export_identifier = row["export_identifier"]
+            except KeyError:
+                export_identifier = cls.make_identifier(
+                    latlons, row["start_date"], row["end_date"]
```

Comment:
why not just pass the bounding_box here, what's the idea behind `latlons`?

Reply:
its for the identifier id - `f"min_lat={min_lat}_min_lon={min_lon}_max_lat={max_lat}_max_lon={max_lon}_dates={start_date}_{end_date}_all"`

I felt easier (?) to pass the latlons along rather than recalculate them from the bounding box.

Reply:
Are they not available as properties of the bounding box?

Reply:
It's a nuisance to interact with earth engine objects, so I prefer not to if I can. For example, rounding errors:
```
>>> import ee
>>> ee.Initialize()
>>> min_lon, max_lon = 1, 2
>>> min_lat, max_lat = 1, 2
>>> poly = ee.Geometry.Polygon([[[min_lon, min_lat], [min_lon, max_lat], [max_lon, max_lat], [max_lon, min_lat]]])
>>> poly.bounds().getInfo()["coordinates"]
[[[0.9999999999999998, 0.9999999999999746], [2, 0.9999999999999746], [2, 2.0000760949199377], [0.9999999999999998, 2.0000760949199377], [0.9999999999999998, 0.9999999999999746]]]
```